pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -74,6 +74,9 @@ func GetPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clien
 		return func(ctx context.Context, name string) (v1beta1.PipelineObject, error) {
 			params := map[string]string{}
 			stringReplacements, arrayReplacements, objectReplacements := paramsFromPipelineRun(ctx, pipelineRun)
+			for k, v := range getContextReplacements("", pipelineRun) {
```

Comment:
Is it possible to get the Pipeline name here? Do all resolvers need a param that is the pipeline name, or can we not make that guarantee?

Reply:
We can't make any guarantee, no, in almost all cases. In practice, the hub or bundles resolver params include what we expect will be the `Pipeline` name, but even then, we can't _guarantee_ it - the actual `Pipeline` could have a different name than its key, e.g. The cluster resolver does know the name, but the git resolver has no idea.

Reply:
Could we add a new field to resolver for Pipeline name instead of passing it in as a param? I would imagine almost all resolvers would need it

Reply:
I don't think we should - we never create an actual `Pipeline`, we just load the spec in to `PipelineRun.Spec.PipelineSpec`...

Reply:
I don't think we should either.

Reply:
That and while the name happens to be a "key" for three of the current resolvers, there's no reason to assume that'll be the case for all or even necessarily a majority of future resolvers. Plus adding another `name`-related field to `pipelineRef`/`taskRef` would be pretty dang confusing.

Reply:
ok, sounds good